File: modules/earth-video/ios/EarthVideoModule.swift
import ExpoModulesCore

public class EarthVideoModule: Module {
  public func definition() -> ModuleDefinition {
    Name("EarthVideo")

    View(EarthVideoView.self) {
      // The `source` prop is a local file URL (file://…) resolved by
      // expo-asset on the JS side before mounting. The view rejects a
      // no-op when the URL is unchanged so React re-renders are cheap.
      Prop("source") { (view: EarthVideoView, url: URL) in
        view.setSource(url)
      }
    }
  }
}

File: modules/earth-video/android/src/main/java/expo/modules/earthvideo/EarthVideoModule.kt
package expo.modules.earthvideo

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Android stub. The session-stealing problem this module solves is
 * iOS-specific (AVPlayer ↔ AVAudioSession election). Android doesn't
 * have an equivalent single-session model — ExoPlayer / MediaPlayer
 * can run a muted background video without claiming the foreground
 * media notification. When we wire Android up, this should become an
 * ExoPlayer view (audio track disabled). For now the JS callers fall
 * back to a static image on Android.
 */
class EarthVideoModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("EarthVideo")

    View(EarthVideoView::class) {
      Prop("source") { _: EarthVideoView, _: String -> }
    }
  }
}
